sdk/src/types.rs

```diff
@@ -516,50 +516,75 @@ pub struct KeylessAccount {
     #[allow(dead_code)]
     pepper: Pepper,
     zk_sig: ZeroKnowledgeSig,
-    jwt: String,
+    jwt_header_json: String,
+    #[allow(dead_code)]
+    jwt: Option<String>,
 }
 
 impl KeylessAccount {
     pub fn new(
-        jwt: String,
+        iss: &str,
+        aud: &str,
+        uid_key: &str,
+        uid_val: &str,
+        jwt_header_json: &str,
         ephemeral_key_pair: EphemeralKeyPair,
         pepper: Pepper,
         zk_sig: ZeroKnowledgeSig,
     ) -> Result<Self> {
-        let parts: Vec<&str> = jwt.split('.').collect();
-        let jwt_payload_json = base64::decode_config(parts[1], base64::URL_SAFE).unwrap();
-        let claims: Claims = serde_json::from_slice(&jwt_payload_json)?;
-
-        let uid_key = "sub".to_owned();
-        let uid_val = claims.get_uid_val(&uid_key)?;
-        let aud = claims.oidc_claims.aud;
-
         let idc = IdCommitment::new_from_preimage(&pepper, &aud, &uid_key, &uid_val)?;
         let public_key = KeylessPublicKey {
-            iss_val: claims.oidc_claims.iss,
+            iss_val: iss.to_owned(),
             idc,
         };
         Ok(Self {
             public_key,
             ephemeral_key_pair,
-            uid_key,
-            uid_val,
-            aud,
+            uid_key: uid_key.to_string(),
+            uid_val: uid_val.to_string(),
+            aud: aud.to_string(),
             pepper,
             zk_sig,
-            jwt,
+            jwt_header_json: jwt_header_json.to_string(),
+            jwt: None,
         })
     }
 
-    pub fn get_jwt_header_json(&self) -> String {
-        let parts: Vec<&str> = self.jwt.split('.').collect();
+    pub fn derive_account(
+        jwt: &str,
+        ephemeral_key_pair: EphemeralKeyPair,
+        pepper: Option<Pepper>,
+        zk_sig: Option<ZeroKnowledgeSig>,
+    ) -> Result<Self> {
+        let parts: Vec<&str> = jwt.split('.').collect();
         let header_bytes = base64::decode(parts[0]).unwrap();
-        String::from_utf8(header_bytes).unwrap()
+        let jwt_header_json = String::from_utf8(header_bytes).unwrap();
+        let jwt_payload_json = base64::decode_config(parts[1], base64::URL_SAFE).unwrap();
+        let claims: Claims = serde_json::from_slice(&jwt_payload_json)?;
+
+        let uid_key = "sub".to_owned();
```

Comment:
i forgot how `uid_key` is determined: isn't it possible for it to be `email` as well?

Reply:
it is.  added an arg to allow for specifying